Book-A-Desk.Api.Tests/FilteredBookingsTest.fs

```diff
@@ -0,0 +1,90 @@
+module Book_A_Desk.Api.Tests.FilteredBookingsTest
```

Comment:
In this project, there are two types of tests:
1. API Tests, which test calling the API directly
2. Unit Tests, which call code directly in an isolated fashion

Currently, the test in this file is kind of a mix of both an API Test and a Unit Test. You are not fully isolating the code (where you would directly call ReservationsQueriesHandler.getFilteredBookings) and you are not calling the API. Let me show you what I mean.

This file is in the API Tests project, so let me show you how to create an API Test. First, we want to setup a TestServer. This TestServer will run an ASP.Net server in-memory and allows you to call the API directly. This allows you to test the integration of the different parts of the system without requiring you to fully run a server. You can see in the file GetBookingsTests.fs in the lines starting with 

``` fsharp
let mockApiDependencyFactory = //...
use httpClient = TestServer.createAndRun mockApiDependencyFactory 
```

What this code does is sets up all of the dependencies of the system (such as the mock provide event store) and directly tests the code all the way from Program.fs to the HttpHandler. The goal is to test the integration of the different parts of the system. Since this might change more often than the rest of the system, we generally write a limited number of API tests, where the goal is to test that all of the code works well together, more than individual parts.

When writing UnitTests, you want to include the files the Domain.Tests project. As well, we generally only unit test the files in the Domain, since this is where the core and complex part of the system should live. When writing unit tests, you provide only what is required to run the test. For ReservationsQueriesHandler.getFilteredBookings, this means a sequence of BookingEvent, a Date Option, OfficeId Option and Email option. How this information is provided (in our code, by the EventStore) is not important. This allows you to test the code very thoroughly since you are not relying on dependencies that could potentially change, making these tests very stable